tests/storage/__init__.py

```diff
@@ -139,70 +109,111 @@ def add_blob(data):
         observation_id=observation_two_first.id,
         response_definition_id=response_definition_two.id,
         active_ref=add_blob([True]),
         update_id=None,
     )
 
-    repository._add_observation_response_definition_link(
+    rdb_api._add_observation_response_definition_link(
         observation_id=observation_two_second.id,
         response_definition_id=response_definition_two.id,
         active_ref=add_blob([True]),
         update_id=None,
     )
-    repository.flush()
 
     ######## add realizations ########
-    realization_0 = repository.add_realization(0, ensemble.name)
-    realization_1 = repository.add_realization(1, ensemble.name)
+    realization_0 = rdb_api.add_realization(0, ensemble.name)
+    realization_1 = rdb_api.add_realization(1, ensemble.name)
     db_lookup["realization_0"] = realization_0.id
 
     def add_data(realization, response_def, ens):
-        response_one = repository.add_response(
+        response_one = rdb_api.add_response(
             name="response_one",
             values_ref=add_blob([11.1, 11.2, 9.9, 9.3]),
             realization_index=realization.index,
             ensemble_name=ens.name,
         )
-        repository.flush()
-        repository._add_misfit(200, obs_res_def_link.id, response_one.id)
+        rdb_api._add_misfit(200, obs_res_def_link.id, response_one.id)
 
-        repository.add_response(
+        rdb_api.add_response(
             name="response_two",
             values_ref=add_blob([12.1, 12.2, 11.1, 11.2, 9.9, 9.3]),
             realization_index=realization.index,
             ensemble_name=ens.name,
         )
 
-        repository.add_parameter(
-            "A", "G", add_blob(1), realization.index, "ensemble_name"
-        )
-        repository.add_parameter(
-            "B", "G", add_blob(2), realization.index, "ensemble_name"
-        )
-        repository.add_parameter(
+        rdb_api.add_parameter("A", "G", add_blob(1), realization.index, "ensemble_name")
+        rdb_api.add_parameter("B", "G", add_blob(2), realization.index, "ensemble_name")
+        rdb_api.add_parameter(
             "key1", "group", add_blob(2), realization.index, "ensemble_name"
         )
 
     add_data(realization_0, response_def=response_definition_one, ens=ensemble)
     add_data(realization_1, response_def=response_definition_one, ens=ensemble)
 
-    repository.commit()
     ######## add blob #########
     data_blob = add_blob([0, 1, 2, 3])
     db_lookup["data_blob"] = data_blob
 
-    yield (db_url, db_lookup)
+    rdb_session.commit()
+    blob_session.commit()
+
+    rdb_session.close()
+    blob_session.close()
+
+    yield db_lookup
+
+
+@pytest.fixture
+def apis(initialize_databases):
+    rdb_session = ERT_STORAGE.RdbSession()
+    blob_session = ERT_STORAGE.BlobSession()
+
+    rdb_api = RdbApi(session=rdb_session)
+    blob_api = BlobApi(session=blob_session)
+
+    try:
+        yield rdb_api, blob_api
+
+        rdb_session.rollback()
```

Comment:
Should rollback regardless, not only during success (move these to after `finally:`)